src/app/store/controller/selectors.ts

```diff
@@ -373,6 +374,34 @@ const getByFabricId = createSelector(
       }, [])
 );
 
+/**
+ * Get all region controllers and separate by their vault configuration status.
+ * @param state - The redux state.
+ * @returns Two lists of region controllers - one where none are configured with vault, the other where all are configured with vault.
+ */
+const getVaultConfiguredControllers = createSelector(
+  [defaultSelectors.all],
+  (controllers: Controller[]) => {
+    const regionControllers = controllers.filter((controller) => {
+      return (
+        controller.node_type === NodeType.REGION_CONTROLLER ||
+        controller.node_type === NodeType.REGION_AND_RACK_CONTROLLER
+      );
+    });
+    const unconfiguredControllers = regionControllers.filter((controller) => {
+      return (
+        controller.vault_configured === false ||
+        controller.vault_configured === undefined
+      );
+    });
+    const configuredControllers = regionControllers.filter((controller) => {
```

Comment:
If still think it's better to use a single selector, might as well do it a single loop, e.g. using reduce (everything that's _not_ a configured controller will be an unconfigured controller)?

Reply:
Same as the other one really - these two return values are always needed together in their implementations, and changing to an object instead of a list would require a lot less refactoring than separating it into separate selectors for configured and unconfigured